### PR TITLE
Add batch prove openings

### PR DESCRIPTION
**Pull request by sai-deng**
Add function `prove_openings` to prove from `FriInstanceInfo`.

### CODE REVIEW

**Review by codeblooded1729 — APPROVED** [review]
Looks great to me. Nice work!

File: plonky2/src/fri/batch_oracle.rs [hunk 1]
```diff
@@ -113,4 +120,279 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
...
+                // Collect the coefficients of all the polynomials in `polynomials`.
+                let polys_coeff = polynomials.iter().map(|fri_poly| {
+                    &oracles[fri_poly.oracle_index].polynomials[fri_poly.polynomial_index]
+                });
+                let composition_poly = timed!(
+                    timing,
+                    &format!("reduce batch of {} polynomials", polynomials.len()),
+                    alpha.reduce_polys_base(polys_coeff)
+                );
+                let mut quotient = composition_poly.divide_by_linear(*point);
```
[the remark this answers is not in the document]
Comment (sai-deng) [reply plonky2/src/fri/batch_oracle.rs:161]: Thanks for the notes.

File: plonky2/src/fri/batch_oracle.rs [hunk 2]
```diff
@@ -113,4 +120,279 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
...
+                },
+                FriBatchInfo {
+                    point: eta,
+                    polynomials: vec![FriPolynomialInfo {
+                        oracle_index: 0,
+                        polynomial_index: 0,
+                    }],
+                },
+            ],
+        };
```
[the remark this answers is not in the document]
Comment (sai-deng) [reply plonky2/src/fri/batch_oracle.rs:295]: I attempted to add degree_bits to the FriInstanceInfo. The issue is that FriInstanceInfo is used in both batch and non-batch versions. In the non-batch version, this addition becomes redundant. Therefore, we need to pass a degree_bits vector along with it, and perform the consistency check in batch_prove_openings().

[the remark this answers is not in the document]
Comment (codeblooded1729) [reply plonky2/src/fri/batch_oracle.rs:296]: Got it. I see that you have used non batch FRI for non public tables, and batch FRI instances for the public tables in the Batch Stark PR. I understand your intention better now.

**Pull request merged**